### PR TITLE
add node Location and node Name on Value metadata

### PR DESCRIPTION
**Pull request by billiaz**
MQTT is a powerfull tool. Using data from to generate graphs is a good approach. 
In tools like influxdb, you rely on the whole Payload read. Adding node Location and Name in metadata is a way to use and analyse the data

Added nodeLocation and nodeName as metadata, to help searching/storing data.

### CODE REVIEW

**Comment by robertsLando** [comment]
@billiaz IMO this should be added in a configuration options like "add node info to payload" (available only for json and valueId payloads). Adding them to valueIds is not a good practice as it will be repeated. It should be added only before sending the payload

**Review by robertsLando — CHANGES_REQUESTED** [review]
Needs to be added only to the payload and make it configurable using options in the ui

**Comment by billiaz** [comment]
> Needs to be added only to the payload and make it configurable using options in the ui

I suspect this https://github.com/zwave-js/zwavejs2mqtt/blob/03d4d0e6cb69a82156f588346ede71572ca85a2a/lib/Gateway.js#L291 is a good point to build the code.

Maybe is a better spot I cannot currently see

**Review by robertsLando — CHANGES_REQUESTED** [review]

File: lib/Gateway.js [hunk 1]
```diff
@@ -291,6 +291,10 @@ function onValueChanged (valueId, node, changed) {
     case 1: // entire zwave valueId object
       data = copy(valueId)
       data.value = tmpVal
+      if (this.config.includeNodeInfo) {
```
[the remark this answers is not in the document]
Comment (billiaz) [reply lib/Gateway.js:294]: I will add this. but just before this, there is a `data.value` which obviously has to work only if this is an object.

maybe is better to  make it
```
if (typeof data === 'object) {
  data.value = tmpVal
  if(this.config.includeNodeInfo) {
    data.nodeName = node.name
    data.nodeLocation = node.loc
  }
}
```

[the remark this answers is not in the document]
Comment (robertsLando) [reply lib/Gateway.js:294]: I mean add this after the switch

[the remark this answers is not in the document]
Comment (billiaz) [reply lib/Gateway.js:294]: I understand, my question is. if data.value assumes data is object, why only check the nodeName/Location in our case?
shouldn't data.value also be checked for the same  typeof?

[the remark this answers is not in the document]
Comment (robertsLando) [reply lib/Gateway.js:294]: nope, in the switch we check for the payload type, later if the payload is an object and the includeNodeInfo flag is true we add the node info

File: lib/Gateway.js [hunk 2]
```diff
@@ -303,6 +300,10 @@ function onValueChanged (valueId, node, changed) {
       data = { time: Date.now(), value: tmpVal }
   }
 
+  if (typeof data === 'object' && this.config.includeNodeInfo) {
```
Comment (robertsLando) [inline lib/Gateway.js:303]:
```suggestion
  if (this.config.includeNodeInfo && typeof data === 'object') {
```

This order makes the check quicker

**Review by robertsLando — CHANGES_REQUESTED** [review]

**Review by robertsLando — APPROVED** [review]
LGTM. Thanks @billiaz :pray:

**Pull request merged**